src/sentry/deletions/defaults/detector.py

```diff
@@ -0,0 +1,10 @@
+from sentry.deletions.base import BaseRelation, ModelDeletionTask, ModelRelation
+from sentry.incidents.models.alert_rule import AlertRule
+from sentry.workflow_engine.models import Detector
+
+
+class DetectorDeletionTask(ModelDeletionTask[Detector]):
+    def get_child_relations(self, instance: AlertRule) -> list[BaseRelation]:
+        from sentry.workflow_engine.models import DataSourceDetector
+
+        return [ModelRelation(DataSourceDetector, {"detector_id": instance.id})]
```

Comment:
lol this is pointless, it already cascade deletes. not sure how to make sure the data condition group and data sources are deleted